docs/DeveloperGuide.md

```diff
@@ -1,38 +1,239 @@
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
-
-## Non-Functional Requirements
-
-{Give non-functional requirements}
-
-## Glossary
-
-* *glossary item* - Definition
-
-## Instructions for manual testing
-
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+---
+title: Developer Guide
+---
+
+* Table of Contents
+{:toc}
+
+# Developer Guide
+
+## Acknowledgements
+
+* The CS2113 Team (Professor and Teaching Assistants) For their guidance in teaching us concepts of Software
+  Engineering and OOP
+
+* Exercise Data retrieved from: [Wrkout](https://github.com/wrkout/exercises.json) (open source MIT license)
+  Used to generate the data.json file containing all the exercises data used
+
+## Design & implementation
+
+### Architecture Component
+
+<div align="center">
+<img src="UML/Images/architecture.png"/>
+<p>
+Figure 1.1
+</p>
+</div>
+
+The architecture diagram as shown in Figure 1.1 shows us the high level design of Fitness Duke.
+
+**Main Components**
+
+The architecture comprises five main components.
+* ```main```: Responsible for the initialisation of user data, app data and connection of various components together
+* ```Ui```: The User Interface of the app
+* ```CommandHandler```: The main logic controller of the app
+* ```Storage```: Reading and Writing data to and from the hard disk
+
+```Hard Drive``` represents the files in which user data are stored
+
+**Interaction of components**
+
+<div align="center">
+<img src="UML/Images/overall_sequence-0.png"/>
+<p>
+Figure 1.2
+</p>
+</div>
+
+The sequence diagram as shown in Figure 1.2 shows the interaction of all the main components when a user enters the 
+```start``` command and subsequently enters the ```finish``` command to simulate the process of a workout session 
+experienced by the user.
+
+The five main components in the diagram shows the main classes participating in the process.
+
+These classes will be defined into its own API as an ```interface``` in the upcoming iteration
+
+More details on these components will be described below.
+
+### UI Component
+
+Represented by the Ui class, it handles the interactions between the user and the program. In addition, it also displays
+a basic overview of what the program does.
+
+Types of pre-defined messages:
+
+1. Information messages that describe the functionalities of the program (`Greet` , `Bye`, `PrintExercises` classes)
+2. Help messages that details the usage of the possible commands available in the program (`PrintHelpMessage` class)
+3. Error messages that trigger when a user's input is incorrect and provides an explanation to the
+   user. (`ErrorMessages` class)
+
+The class diagram as shown in Figure *1.3* illustrates the structure of the different classes in Ui.
+
+<div align="center">
+<img src="UML/Images/Ui.png"/>
+<p>
+Figure 1.3

```

Comment:
There can be more comprehensive explanation to the diagram given the diagram's complexity